torch/distributed/fsdp/wrap.py

```diff
@@ -209,6 +211,48 @@ def __repr__(self) -> str:
         return super().__repr__() + f"({self._module_classes_str})"
 
 
+class LambdaWrapPolicy(_WrapPolicy):
```

Comment:
You solicited some name bikeshedding over WC, so I'll try my best here. Assuming we are keeping WrapPolicy suffix, CallbackWrapPolicy or CustomWrapPolicy both convey intent here reasonably well.

Reply:
You could also try PredicateWrapPolicy. This is not a perfect match: I would call `lambda_fn` a predicate if it only returned a bool, but you return a little bit extra information sometimes.